modules/tftp/server.rb

```diff
@@ -150,6 +150,21 @@ def pxeconfig_file(mac)
     end
   end
 
+  def self.fetch_boot_image(dst, src)
+
+    # Verify dst is a valid directory
+    dst_file = Pathname.new(dst).cleanpath
+    dst_path = Pathname.new(dst.delete_suffix(".iso"))
+
+    FileUtils.mkdir_p dst_path.parent
```

Comment:
What if the parent is already present? Will you overwrite files?

Reply:
As far as the directory is concerned, this is handled by `mkdir_p`, which `causes to make following directories, if it does not exist.` (https://ruby-doc.org/stdlib-2.5.1/libdoc/fileutils/rdoc/FileUtils.html#method-c-mkdir_p)

The `dst_file` itself will be overwritten, I guess.

Reply:
> The dst_file itself will be overwritten, I guess.

Will you have the case where 2 concurrent hosts are provisioning and one reads the file while it's being overwritten?

Reply:
Fixed with implementing the locking from the `http_download` module.